pkg/agent/flowexporter/connections/conntrack_connections_perf_test.go

```diff
@@ -75,18 +80,33 @@ func BenchmarkPoll(b *testing.B) {
 	b.Logf("\nSummary:\nNumber of initial connections: %d\nNumber of new connections/poll: %d\nNumber of deleted connections/poll: %d\n", testNumOfConns, testNumOfNewConns, testNumOfDeletedConns)
 }
 
-func setupConntrackConnStore(b *testing.B) (*ConntrackConnectionStore, *connectionstest.MockConnTrackDumper) {
+func setupConntrackConnStore(b *testing.B, connections []*flowexporter.Connection) (*ConntrackConnectionStore, *connectionstest.MockConnTrackDumper) {
 	ctrl := gomock.NewController(b)
 	defer ctrl.Finish()
-	mockIfaceStore := interfacestoretest.NewMockInterfaceStore(ctrl)
-	testInterface := &interfacestore.InterfaceConfig{
-		Type: interfacestore.ContainerInterface,
-		ContainerInterfaceConfig: &interfacestore.ContainerInterfaceConfig{
-			PodName:      "pod",
-			PodNamespace: "pod-ns",
+	k8sClient := fake.NewSimpleClientset()
+	podInformer := coreinformers.NewPodInformer(k8sClient, metav1.NamespaceAll, 0, cache.Indexers{})
+	podStore := podstore.NewPodStore(podInformer)
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+	go podInformer.Run(stopCh)
+	cache.WaitForCacheSync(stopCh, podInformer.HasSynced)
+	pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "pod-ns",
+			Name:      "pod",
+			UID:       "pod",
 		},
+		Status: v1.PodStatus{
+			Phase: v1.PodPending,
+		},
+	}
+	for _, conn := range connections {
+		pod.Status.PodIPs = append(pod.Status.PodIPs, v1.PodIP{IP: conn.FlowKey.SourceAddress.String()}, v1.PodIP{IP: conn.FlowKey.DestinationAddress.String()})
 	}
-	mockIfaceStore.EXPECT().GetInterfaceByIP(gomock.Any()).Return(testInterface, true).AnyTimes()
+	k8sClient.CoreV1().Pods(pod.Namespace).Create(context.TODO(), pod, metav1.CreateOptions{})
+	assert.Eventuallyf(b, func() bool {
+		return len(podInformer.GetIndexer().List()) == 1
+	}, 1*time.Second, 10*time.Millisecond, "Pods should be added to podInformer")
```

Comment:
looking at this, I feel like the test got more complicated for no good reason. I think we could keep the same level of abstraction as before by defining an interface for the `PodStore`, that only exposes the `GetPodByIPAndTime` method. We can then mock the interface and keep these FE / FA tests simpler.

Reply:
That make sense to me. I've created a mockInterface and applied it to all the tests.